File: codes/greedy/53.maximum-subarray/maximum-subarray.cpp
#include <vector>
#include <algorithm>

using namespace std;
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int result = INT_MIN;
        int count = 0;
        for (auto& i : nums) {
            count += i;

            // 一定要放到前面
            result = max(result, count);
            if(count<0){
                count = 0;
            }

        }
        return result;
    }
    };

File: codes/greedy/53.maximum-subarray/maximum-subarray.go
package main
func maxSubArray(nums []int) int {
	result:=math.MinInt
	count:=0
	for _,i:=range nums{
		count+=i
		result=max(result,count)
		if count<0{
			count=0
		}

	}
	return result

}

func max(a,b int)int{
	if a>=b{
		return a
	}
	return b
}